include/SharedMemory.h

```diff
@@ -71,6 +74,25 @@ class SharedMemoryData
 	void* m_ptr = nullptr;
 };
 
+	static inline std::string createKey() {
+#if defined(LMMS_BUILD_APPLE) || defined(LMMS_BUILD_FREEBSD) || defined(LMMS_BUILD_OPENBSD)
+		constexpr int length = 30; // PSHMNAMLEN=31
```

Comment:
It may be desired by the project to just hard-code `30` here and skip the `#ifdef`, as a collision would be nearly impossible, however the comment may help other use of the class down the road.

Reply:
How does this look for all OS's?
```cpp
// Max length (minus prepended '/') on macOS (PSHMNAMLEN=31)
constexpr int length = 30;
```

Reply:
Readable enough for me!  I thought that I had read somewhere that all BSD flavors had that limitation but I couldn't confirm this earlier and added them out of paranoia.  I think it's good!